mail/tasks.py

```diff
@@ -0,0 +1,75 @@
+"""
+Tasks for mail app
+"""
+import logging
+from decimal import Decimal
+
+from django.conf import settings
+from rest_framework import status
+
+from bootcamp.celery import async
+from ecommerce.models import Line
+from klasses.models import Klass, BootcampAdmissionCache
+from mail.api import MailgunClient
+from mail.models import AutomaticReminderEmail, SentAutomaticEmails
+from mail.utils import generate_mailgun_response_json
+
+log = logging.getLogger(__name__)
+
+
+def send_reminder_payment_emails():
+    """
+    Checks if there are klasses that have an installment coming and
+    sends an email to all the users who need to be reminded of paying
+    if the deadline is in X days (where X is defined in the AutomaticReminderEmail)
+    """
+    reminder_email_in_days = list(AutomaticReminderEmail.objects.filter(
+        reminder_type=AutomaticReminderEmail.PAYMENT_APPROACHING).values_list('days_before', flat=True))
+    klasses_reminder = []
+    for klass in Klass.objects.all():
+        if klass.next_payment_deadline_days in reminder_email_in_days:
+            klasses_reminder.append(klass)
+    if not klasses_reminder:
+        return
+
+    for klass in klasses_reminder:
+        email_template = AutomaticReminderEmail.objects.filter(days_before=klass.next_payment_deadline_days).first()
+        all_admissions = BootcampAdmissionCache.objects.filter(klass=klass)
+        for admission in all_admissions:
+            user = admission.user
+            if SentAutomaticEmails.objects.filter(user=user, automatic_email=email_template, klass=klass).exists():
+                # reminder already sent
+                continue
+            total_already_paid = Line.total_paid_for_klass(user, klass.klass_key).get('total') or Decimal('0.00')
+            remaining_balance = klass.total_due_by_next_deadline - total_already_paid
+            if remaining_balance <= 0:
+                # amount already paid
+                continue
+            response = MailgunClient.send_individual_email(
```

Comment:
There may still be exceptions raised even given `raise_for_status=False`, for example if we fail to contact mailgun due to some network outage. You might want to wrap this in a try block and set `raise_for_status=True`